File: fabric/src/main/java/juuxel/adorn/resources/AdornResources.java
package juuxel.adorn.resources;

import juuxel.adorn.client.resources.BookManagerFabric;
import juuxel.adorn.client.resources.ColorManagerFabric;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.fabricmc.fabric.api.resource.ResourceManagerHelper;
import net.minecraft.server.packs.PackType;

public final class AdornResources {
    @Environment(EnvType.CLIENT)
    public static void initClient() {
        var resourceManagerHelper = ResourceManagerHelper.get(PackType.CLIENT_RESOURCES);
        resourceManagerHelper.registerReloadListener(ColorManagerFabric.INSTANCE);
        resourceManagerHelper.registerReloadListener(BookManagerFabric.INSTANCE);
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/resources/AdornResources.kt
package juuxel.adorn.resources

import juuxel.adorn.AdornCommon
import juuxel.adorn.client.resources.BlockVariantTextureLoader
import juuxel.adorn.client.resources.BookManagerFabric
import juuxel.adorn.client.resources.ColorManagerFabric
import juuxel.adorn.platform.fabric.ResourceReloaderWithId
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.fabric.api.resource.ResourceManagerHelper
import net.minecraft.resource.ResourceType

object AdornResources {
    @Environment(EnvType.CLIENT)
    fun initClient() {
        ResourceManagerHelper.get(ResourceType.CLIENT_RESOURCES).apply {
            registerReloadListener(ColorManagerFabric)
            registerReloadListener(BookManagerFabric)
            registerReloadListener(ResourceReloaderWithId(AdornCommon.id("block_variant_textures"), BlockVariantTextureLoader))
        }
    }
}
